app/models/concerns/browseable.rb

```diff
@@ -0,0 +1,157 @@
+module Browseable
+  extend ActiveSupport::Concern
+
+  included do
+    class_attribute :facets
+    self.facets = {}
+  end
+
+  class Facets
+    include Enumerable
+
+    attr_reader :search, :klass
+
+    def initialize(search)
```

Comment:
This could take ``klass`` instead of ``search``, we don't seem to use ``search`` other than to call ``.klass`` on it.